api/src/Downstream/Type/FastEp.php

```diff
@@ -17,82 +17,98 @@ function results() {
         if ($pdb) {
             if (file_exists($pdb["FILE"])) {
                 $pdb = file_get_contents($pdb["FILE"]);
-                foreach (explode("\n", $pdb) as $l) {
-                    if (strpos($l, 'HETATM') !== false) {
-                        $parts = preg_split('/\s+/', $l);
-                        array_push($ats, array(
-                            $parts[1],
-                            $parts[5],
-                            $parts[6],
-                            $parts[7],
-                            $parts[8],
-                        ));
-                    }
-                }
+            } elseif (file_exists($pdb['FILE'].'.gz')) {
+                $zd = gzopen($pdb['FILE'].'.gz', 'r');
+                $pdb = gzread($zd, 10000000);
+                gzclose($zd);
+            } else {
```

Comment:
This is limited to 10^7 bytes of uncompressed data, which might not be enough for some files. An alternative data stream/loading strategy should be considered.